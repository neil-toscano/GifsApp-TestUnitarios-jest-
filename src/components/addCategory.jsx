import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('');
  const handleInputChange = (event) => {
    //cambiamos el valor del input
    setInputValue(event.target.value);
  };
  const onsubmit = (event) => {
    //al darle enter en el input
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;

    // setCatego((data)=>[...data,inputValue]);
    onNewCategory(inputValue.trim());
    setInputValue('');
  };

  return (
    <form onSubmit={(event) => onsubmit(event)} aria-label='form'>
      <input
        placeholder="Buscar..."
        type="text"
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
