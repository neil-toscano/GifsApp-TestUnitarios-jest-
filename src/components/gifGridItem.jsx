import PropTypes from 'prop-types';

export const GifItem = ({ titulo, url }) => {
  return (
    <div className="card">
      <img src={url} alt={titulo} />
      <p>{titulo}</p>
    </div>
  );
};

GifItem.propTypes = {
  titulo: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

GifItem.defaultProps = {
  titulo: 'Gifs',
  url: 'https://pixels.com',
};
