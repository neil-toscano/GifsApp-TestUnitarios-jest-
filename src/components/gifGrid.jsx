import { GifItem } from './gifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {isLoading  ? <h2>Cargando...</h2> : null}
      <div className="card-grid">
        {images.map(({ id, titulo, url }) => (
          <GifItem key={id} titulo={titulo} url={url} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};

