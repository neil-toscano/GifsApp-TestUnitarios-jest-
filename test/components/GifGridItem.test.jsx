import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/gifGridItem';

describe('pruebas GifGridItem', () => {
  const titulo = 'one Punch';
  const url =
    'https://media4.giphy.com/media/Qm3xCXWifrgzb9aEqg/giphy.gif?cid=ae0bb4fawd335f9tdt7owa3rdpeg3zltwpmroo4cc8k39b3a&ep=v1_gifs_search&rid=giphy.gif&ct=g';
  
    test('debe hacer match con el snapshot', () => {
    const { container } = render(<GifItem titulo={titulo} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('debe de mostrar la img con el URL y el ALT indicado', () => { 
    render(<GifItem titulo= {titulo} url= {url} />);

    // expect(screen.getByRole('img').src).toBe(url);
    // expect(screen.getByRole('img').alt).toBe(titulo);
    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  })

  test('debe de mostrar el tituo en el componente', () => { 
    render(<GifItem titulo={titulo} url={url} />);
    expect(screen.getByText(titulo)).toBeTruthy();
   })
});
