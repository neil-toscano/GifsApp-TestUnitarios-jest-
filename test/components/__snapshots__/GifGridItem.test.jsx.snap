// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas GifGridItem debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="one Punch"
      src="https://media4.giphy.com/media/Qm3xCXWifrgzb9aEqg/giphy.gif?cid=ae0bb4fawd335f9tdt7owa3rdpeg3zltwpmroo4cc8k39b3a&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      one Punch
    </p>
  </div>
</div>
`;
